### PR TITLE
Define _PyCFunctionFastWithKeywords() on CPython 3.7+

### PR DESCRIPTION
**Pull request by ijl**
https://docs.python.org/3/c-api/structures.html#c._PyCFunctionFastWithKeywords

### CODE REVIEW

**Comment by ijl** [comment]
Regarding the 3.7 cfg, this was introduced in cpython commit 6969eaf4682beb01bc95eeb14f5ce6c01312e297 and first tagged in v3.7.0a1.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Added
 - Add unsafe API `with_embedded_python_interpreter` to initalize a Python interpreter, execute a closure, and finalize the interpreter. [#1355](https://github.com/PyO3/pyo3/pull/1355)
 - Add `serde` feature to support `Serialize/Deserialize` for `Py<T>`. [#1366](https://github.com/PyO3/pyo3/pull/1366)
+- Define `_PyCFunctionFastWithKeywords()` on CPython 3.7+. [#1384](https://github.com/PyO3/pyo3/pull/1384)
```
Comment (davidhewitt) [inline CHANGELOG.md:12]: Just a small nit to make this wording match other ones in the changelog:

```suggestion
- Add FFI definition `_PyCFunctionFastWithKeywords` on Python 3.7 and up. [#1384](https://github.com/PyO3/pyo3/pull/1384)
```

**Review by davidhewitt — APPROVED** [review]
Thanks for this!

I've got a tiny nit on the changelog and this is good to merge. Looking at the docs you linked I noticed there's a couple of other mismatches we have:
- we incorrectly use `args: *mut *mut PyObject` in `_PyCFunctionFast` when it should also be `*const *mut PyObject` like you have used here.
- we're also missing `PyCMethod` for Python 3.9 and up.

This PR is good to merge with the suggested changelog entry. However, if you've got a moment and willing to fix the above points at the same time I'd be very grateful! 👍

**Comment by nw0** [comment]
>  we incorrectly use `args: *mut *mut PyObject` in `_PyCFunctionFast` when it should also be `*const *mut PyObject` like you have used here.

This is actually correct for 3.6:

```c
typedef PyObject *(*_PyCFunctionFast) (PyObject *self, PyObject **args,
                                       Py_ssize_t nargs, PyObject *kwnames);
```

File: src/ffi/methodobject.rs [hunk 2]
```diff
@@ -30,6 +30,14 @@ pub type PyCFunctionWithKeywords = unsafe extern "C" fn(
     kwds: *mut PyObject,
 ) -> *mut PyObject;
 
+#[cfg(all(Py_3_7, not(Py_LIMITED_API)))]
```
Comment (nw0) [inline src/ffi/methodobject.rs:33]: In `methodobject.h`, this is unconditionally defined (i.e. part of the limited API). (It should probably be considered CPython-specific, however)

```suggestion
#[cfg(all(Py_3_7))]
```

**Comment by davidhewitt** [comment]
👍 thanks for the review @nw0 - in which case I'll merge this now and we can leave further corrections for #1289

**Comment by ijl** [comment]
I see you committed the rewording, thanks.

**Pull request merged**